iroh/src/commands/sync.rs

```diff
@@ -0,0 +1,263 @@
+        match self {
+            Author::List => {
+                let mut stream = iroh.list_authors().await?;
+                while let Some(author_id) = stream.try_next().await? {
+                    println!("{}", author_id);
+                }
+            }
+            Author::Create => {
+                let author_id = iroh.create_author().await?;
+                println!("{}", author_id);
+            }
+        }
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone, Parser)]
+pub enum Docs {
+    List,
+    Create,
+    // Import {
+    //     key: String,
+    //     #[clap(short, long)]
+    //     peers: Vec<PeerSource>,
+    // },
+    Import { ticket: DocTicket },
+}
+
+impl Docs {
+    pub async fn run(self, iroh: Iroh) -> anyhow::Result<()> {
+        match self {
+            Docs::Create => {
+                let doc = iroh.create_doc().await?;
+                println!("created {}", doc.id());
+            }
+            // Docs::Import { key, peers } => {
+            //     let key = hex::decode(key)?
+            //         .try_into()
+            //         .map_err(|_| anyhow!("invalid length"))?;
+            //     let ticket = DocTicket::new(key, peers);
+            //     let doc = iroh.import_doc(ticket).await?;
+            //     println!("imported {}", doc.id());
+            // }
+            Docs::Import { ticket } => {
+                let doc = iroh.import_doc(ticket).await?;
+                println!("imported {}", doc.id());
+            }
+            Docs::List => {
+                let mut stream = iroh.list_docs().await?;
+                while let Some(id) = stream.try_next().await? {
+                    println!("{}", id)
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone, Parser)]
+pub enum Doc {
+    StartSync {
+        peers: Vec<PeerSource>,
+    },
+    Share {
+        mode: ShareMode,
+    },
+    /// Set an entry
+    Set {
+        /// Author of this entry.
+        author: AuthorId,
+        /// Key to the entry (parsed as UTF-8 string).
+        key: String,
+        /// Content to store for this entry (parsed as UTF-8 string)
+        value: String,
+    },
+    /// Get entries by key
+    ///
+    /// Shows the author, content hash and content length for all entries for this key.
+    Get {
+        /// Key to the entry (parsed as UTF-8 string).
+        key: String,
+        /// If true, get all entries that start with KEY.
+        #[clap(short, long)]
+        prefix: bool,
+        /// Filter by author.
+        #[clap(short, long)]
+        author: Option<AuthorId>,
+        /// If true, old entries will be included. By default only the latest value for each key is
+        /// shown.
+        #[clap(short, long)]
+        old: bool,
+
+        /// Also print the content for each entry (but only if smaller than 1MB and valid UTf-8)
+        #[clap(short, long)]
+        content: bool,
+    },
+    List {
```

Comment:
this should probably have the option to filter by author?

Reply:
Added